lib/https.js

```diff
@@ -31,9 +31,15 @@ const inherits = util.inherits;
 const debug = util.debuglog('https');
 const { urlToOptions, searchParamsSymbol } = require('internal/url');
 
-function Server(opts, requestListener) {
+function Server(opts = {}, requestListener) {
   if (!(this instanceof Server)) return new Server(opts, requestListener);
 
+  if (typeof opts === 'function') {
+    requestListener = opts;
+    opts = undefined;
```

Comment:
Then I think you will make the opts default value be unset as well for performance as well. BTW perf is not that crucial here than consistence of readability, this constructor needn't be called too much times in any of applications :)